File: CodeForces/900/SendingMessage.cpp
#include <bits/stdc++.h>
using namespace std;

#ifndef ONLINE_JUDGE
#include "algo/debug.h"
#else
#define dbg(...) ;
#define debug(...) ;
#endif

void solve() {
    
    int n, f, a, b;
    cin >> n >> f >> a >> b;
    
    vector<int> v(n);
    for (int i = 0; i < n; i++) cin >> v[i];
    
    int last = 0;
    
    for (int i = 0; i < n; i++) {
        long long keepOn = 1LL * a * (v[i] - last);
        long long turnOffAndOn = b;
        long long required = min(keepOn, turnOffAndOn);
        
        f -= required;
        last = v[i];
        
        if (f <= 0) {
            cout << "NO" << endl;
            return;
        }
    }
    
    cout << "YES" << endl;
    
}

int main() {
    ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

	int t = 1;
	cin >> t;
	
	for (int tc = 1; tc <= t; tc++) {
		debug(Testcase, tc);
		solve();
		dbg();
	}
	
	return 0;
}

File: CodeForces/900/SendingMessage.java
package Beginner900;

import java.util.Scanner;
import java.util.Vector;


public class SendingMessage {
    static Scanner sc = new Scanner(System.in);

    public static void solve() {
        int n = sc.nextInt();
        int f = sc.nextInt();
        int a = sc.nextInt();
        int b = sc.nextInt();

        Vector<Integer> v = new Vector<>(n);
        for (int i = 0; i < n; i++) {
            v.add(sc.nextInt());
        }
        int last = 0;
        for (int i = 0; i < n; i++) {
            long keepOn = 1L * a * (v.get(i) - last);
            long turnOffAndOn = 0;
            long required = Math.min(keepOn, turnOffAndOn);

            f -= required;
            last = v.get(i);
            if (f <= 0) {
                System.out.println("NO");
            }
        }
        System.out.println("YES");
    }

    public static void main(String[] args) {
        int t = sc.nextInt();
        while (t-- > 0) {
            solve();
        }
    }
}